ports/esp32s2/common-hal/alarm/SleepMemory.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 microDev
+ * Copyright (c) 2020 Dan Halbert for Adafruit Industries
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <string.h>
+
+#include "py/runtime.h"
+#include "common-hal/alarm/SleepMemory.h"
+
+#include "esp_sleep.h"
+
+void alarm_sleep_memory_reset(void) {
+    // Power RTC slow memory during deep sleep
+    esp_sleep_pd_config(ESP_PD_DOMAIN_RTC_SLOW_MEM, ESP_PD_OPTION_ON);
```

Comment:
Please note that an explicit call to `esp_sleep_pd_config(.., ESP_PD_OPTION_ON)` [is not required](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/sleep_modes.html#power-down-of-rtc-peripherals-and-memories):

> If some variables in the program are placed into RTC slow memory (for example, using RTC_DATA_ATTR attribute), RTC slow memory will be kept powered on by default. This can be overridden using esp_sleep_pd_config() function, if desired.

(whether any variables have been placed into RTC_SLOW is determined by the size of corresponding linker section.)

On the other hand, calling `esp_sleep_pd_config(ESP_PD_DOMAIN_RTC_SLOW_MEM, ESP_PD_OPTION_OFF)` here when cpy sleep memory feature is not used has a chance of breaking some other feature which depends on RTC_SLOW memory.

Therefore I'd recommend using the default behavior — not powering RTC_SLOW on or off. The current consumption difference is indeed minimal (0.5uA).

Reply:
@igrr Thanks, I was wondering putting something in that section would trigger making sure the power was on.

Reply:
@igrr If we used RTC_FAST_MEM, is this also still automatic? The [ESP32S2 doc](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s2/api-reference/system/sleep_modes.html#power-down-of-rtc-peripherals-and-memories)  does not mention keeping the RTC_FAST_MEM powered up (the link above is for ESP32). Thanks.

Reply:
Having looked at the code, it seems that RTC_FAST is simply powered on by default. It used to be so because on ESP32 revision 0 we would always need to enable deep sleep wake stub, to work around a hardware bug. Then we didn't change this for the ESP32-S2 in v4.2 to reduce chance of breaking someone's code.

I think we might need to wait until the next major release (v5.0) before we can start automatically powering down RTC_FAST memory if nothing is placed there.